### PR TITLE
Add optional flag to WaitForComponentProperty [C# and Java]

### PR DESCRIPTION
**Pull request by gomboskriszta15**

### CODE REVIEW

File: Assets/AltTester/Runtime/AltDriver/Commands/FindObjects/AltWaitForComponentProperty.cs [hunk 1]
```diff
@@ -29,16 +31,18 @@ public class AltWaitForComponentProperty<T> : AltBaseFindObjects
         string propertyName;
         T propertyValue;
         string assembly;
+        bool getPropertyAsString;
         double timeout;
         double interval;
 
-        public AltWaitForComponentProperty(IDriverCommunication commHandler, string componentName, string propertyName, T propertyValue, string assemblyName, double timeout, double interval, AltObject altObject) : base(commHandler)
+        public AltWaitForComponentProperty(IDriverCommunication commHandler, string componentName, string propertyName, T propertyValue, string assemblyName, bool getPropertyAsString, double timeout, double interval, AltObject altObject) : base(commHandler)
```
Comment (rucindrea) [inline Assets/AltTester/Runtime/AltDriver/Commands/FindObjects/AltWaitForComponentProperty.cs:38]: We should add the new parameter as the last one, so we don't mess up the calls for people that are using the method without explicit parameters - in these cases, their `timeout` param would be interpreted as the `getPropertyAsString`. This applies everywhere I guess

File: Assets/Examples/Test/Editor/Driver/Testforscene1TestSample.cs [hunk 2]
```diff
@@ -237,7 +238,7 @@ public void TestWaitForComponentPropertyTimeOut()
             const string propertyName = "InstrumentationSettings.AltServerPort";
             var altElement = altDriver.FindObject(By.NAME, "AltTesterPrefab");
             Assert.NotNull(altElement);
-            Assert.Throws<WaitTimeOutException>(() => altElement.WaitForComponentProperty(componentName, propertyName, "Test", "AltTester.AltTesterUnitySDK", 2));
+            Assert.Throws<WaitTimeOutException>(() => altElement.WaitForComponentProperty(componentName, propertyName, "Test", "AltTester.AltTesterUnitySDK", timeout: 2));
```
Comment (rucindrea) [inline Assets/Examples/Test/Editor/Driver/Testforscene1TestSample.cs:241]: this is exactly the problem that I mentioned above... we shouldn't have to change a test like this one.

**Review by rucindrea — CHANGES_REQUESTED** [review]
This PR also needs changes to the documentation, can we make them in this one as well?

File: Assets/AltTester/Runtime/AltDriver/Commands/FindObjects/AltWaitForComponentProperty.cs (hunk 1 above)
Comment (gomboskriszta15) [reply Assets/AltTester/Runtime/AltDriver/Commands/FindObjects/AltWaitForComponentProperty.cs]: Done. Related to the docs part, the changes are made here https://github.com/alttester/AltTester-Unity-SDK/pull/1628

**Review by rucindrea — APPROVED** [review]

**Review by andreion1ca — APPROVED** [review]

**Pull request merged**